wormhole-connect/src/hooks/useExternalSearch.ts

```diff
@@ -0,0 +1,50 @@
+import { ChainName } from '@wormhole-foundation/wormhole-connect-sdk';
+import { coalesceChainName } from '@certusone/wormhole-sdk';
+import { CHAINS_ARR, SEARCH_TX } from 'config';
+import { useEffect, useState } from 'react';
+
+const VALID_CHAINS = CHAINS_ARR.filter(
+  (chain) => chain.key !== 'wormchain',
+).map((chain) => chain.key);
+
+type ExternalSearch = {
+  hasExternalSearch: boolean;
+  txHash?: string;
+  chainName?: ChainName;
+  clear: () => void;
+};
+
+export function useExternalSearch(): ExternalSearch {
+  const [hasExternalSearch, setHasExternalSearchtate] =
+    useState<boolean>(false);
+  const [txHash, setTxHash] = useState<string>();
+  const [chainName, setChainName] = useState<ChainName>();
+  useEffect(() => {
+    if (SEARCH_TX?.chainName && SEARCH_TX?.txHash) {
+      const chainName = coalesceChainName(
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        SEARCH_TX.chainName as any,
```

Comment:
there is a mix of types, from on side, there is mix between ChainName from wormhole-sdk and connect-sdk and also it could be a number or an string as chainName and need multiple casting, I've opted in to make it as any